aten/src/ATen/native/Unique.cpp

```diff
@@ -0,0 +1,44 @@
+// Returns unique elements of input tensor.
+
+#include "ATen/ATen.h"
+
+#include <unordered_map>
+#include <unordered_set>
+
+namespace at {
+namespace native{
+
+std::tuple<Tensor, Tensor> unique(
+    const Tensor& self, const bool sorted, const bool return_inverse) {
+  std::unordered_set<int64_t> set(
+      self.data<int64_t>(), self.data<int64_t>() + self.numel());
+  Tensor output = self.type().tensor({static_cast<long long>(set.size())});
```

Comment:
I don't think we should use an `unordered_set` here. The `!sorted` case below has no guarantee that `set.begin()` and `set.end()` is ordered in a way. 

In fact, why do you need a set? Can't you just use torch's sort (if needed)? Then just iterate through once will be enough. Also it gives indices so computing `inverse_indices` should also be convenient.

Reply:
The `!sorted` case doesn't require ordering I think?
Also this set method is `O(N)` (with `O(N)` space) while sorting is `O(NlogN)` so slower complexity wise (if anyone cares)

Reply:
Oh right my bad. I thought misunderstood `sorted` arg. I thought it was saying that the input is sorted.